openstack/placement/resourceproviders/results.go

```diff
@@ -0,0 +1,38 @@
+package resourceproviders
+
+import "github.com/gophercloud/gophercloud/pagination"
+
+type ResourceProviderLinks struct {
+	Href string `json:"href"`
+	Rel  string `json:"rel"`
+}
+
+// ResourceProvider are entities which provider consumable inventory of one or more classes of resource
+type ResourceProvider struct {
+	Generation         int                     `json:"generation"`
+	UUID               string                  `json:"uuid"`
+	Links              []ResourceProviderLinks `json:"links"`
+	Name               string                  `json:"name"`
+	ParentProviderUUID string                  `json:"parent_provider_uuid"`
+	RootProviderUUID   string                  `json:"root_provider_uuid"`
```

Comment:
It looks like the above two fields are part of a microversion?

https://github.com/openstack/placement/blob/master/placement/handlers/resource_provider.py#L56-L58

According to our [microversion rules](https://github.com/gophercloud/gophercloud/blob/master/docs/MICROVERSIONS.md#new-response-fields), these should have their own `Extract` methods. However, Placement is still a relatively new service and I think 1.14 should be an early enough release that we can consider that part of "base".

It might be worthwhile to add a comment about this, though:

```go
// ParentProviderUUID contains the UUID of the parent provider.
// Requires microversion 1.14 or above.
ParentProviderUUID string `json:"parent_provider_uuid"`

// RootProviderUUID contains the UUID of the root provider.
// Requires microversion 1.14 or above.
ParentProviderUUID string `json:"parent_provider_uuid"`
```

Reply:
Sure, I'll add the comments.